main.cpp

```diff
@@ -337,12 +347,21 @@ class Mod : GenericMod {
 		if (creature->entity_data.hostility_type != cube::Creature::EntityBehaviour::Player &&
 			creature->entity_data.hostility_type != cube::Creature::EntityBehaviour::Pet)
 		{
-			*stat /= creature->entity_data.level + 1;
-			*stat += m_CreatureScaling.at(type) * (GetCreatureLevel(creature) - LEVELS_PER_REGION / 2);
-			*stat *= creature->entity_data.level + 1;
+			//*stat /= creature->entity_data.level + 1;
+			//*stat += m_CreatureScaling.at(type) * (GetCreatureLevel(creature) - LEVELS_PER_REGION / 2);
+			//*stat *= creature->entity_data.level + 1;
+			if (type != STAT_TYPE::HASTE && type != STAT_TYPE::CRIT) {
```

Comment:
I would suggest removing the function calls to this function from the `OnCreatureHasteCalculated` and `OnCreatureCritCalculated` functions instead of this. Do note that there is 0 haste and crit scaling currently

Reply:
You are currently also scaling every stat the same way for mobs and not ever using the `m_CreatureScaling` settings. If you were to use that as a multiplier, the whole if statement could be removed even.